websiteapp/views.py

```
@@ -0,0 +1,123 @@
+        source_language = request.POST.get('source')
+        text = request.POST.get('text')
+        if not text or text == "":
+            return JsonResponse({translator: "" for translator in passed_translators})
+        if not source_language:
+            return JsonResponse({'error': 'Source language required : fill with "auto" if you don\'t know...'})
+
+        result = {}
+        for translator in passed_translators:
+            translation = translate_by_translator(request, translator, source_language, target_language, text)
+            if translation is not None:
+                result[translator] = translation
+        return JsonResponse(result)
+
+
+def translate_by_translator(request, translator, source_language, target_language, text):
+    try:
+        # Default translator is google
+        if not translator:
+            translator = 'google'
+
+        if not translators[translator]:
+            return JsonResponse({'error': 'incorrect translator'})
+
+        params_to_give = [source_language]
+        if target_language is not None:
+            params_to_give.append(target_language)
+        for attr in translators[translator]["requiredAttr"] or []:
+            cur_atr = request.POST.get(attr)
+            if not cur_atr:
+                env_param = config.get(translator.upper() + "_" + attr.upper())
+                if env_param is None or env_param == "":
+                    return
+                else:
+                    if env_param not in params_to_give:
+                        params_to_give.append(env_param)
+            else:
+                if cur_atr not in params_to_give:
+                    params_to_give.append(cur_atr)
+
+        target_translator = translators[translator]["translator"](*params_to_give)
+        translation = target_translator.translate(text)
+
+        return translation
+    except ElementNotFoundInGetRequest or RequestError or Exception:
+        return
```

Why just return? In this case we will never know what s going wrong. I tried to test the website locally but it didnt work. Maybe it would be better to throw the exception and to add logging